src/ddsrt/include/dds/ddsrt/events.h

```diff
@@ -0,0 +1,179 @@
+  /**< resource has been opened */
+//#define DDSRT_EVENT_FLAG_OPEN (1u<<2) 
+  /**< resource has been closed */
+//#define DDSRT_EVENT_FLAG_CLOSE (1u<<3) 
+  /**< timeout has occurred on the resource */
+//#define DDSRT_EVENT_FLAG_TIMEOUT (1u<<4)
+  /**< notification of ip address change on the resource*/
+//#define DDSRT_EVENT_FLAG_IP_CHANGE (1u<<5)  
+///@}
+
+  /**
+  * @brief Describes the type of object being monitored for events.
+  */
+  enum ddsrt_event_type {
+    ddsrt_event_type_unset, /**< unitialized state*/
+    ddsrt_event_type_socket /**< indicating a socket type connection*/
+    /*,
+    ddsrt_event_type_file,
+    ddsrt_event_type_ifaddr*/
+  };
+  typedef enum ddsrt_event_type ddsrt_event_type_t;
+
+  /**
+  * @brief Structure containing the information of which object is being watched.
+  *
+  * Is used by ddsrt_event_queue to keep track of objects being watched.
+  */
+  struct ddsrt_event {
+    uint32_t flags; /**< Type of event being watched for, composited from DDSRT_EVENT_FLAG_$EVENT_TYPE$.*/
+    ddsrt_event_type_t type; /**< Type of object being watched.*/
+    ddsrt_atomic_uint32_t triggered; /**< Trigger status after a call to ddsrt_event_queue_wait.*/
+    union {
+      struct { ddsrt_socket_t sock; } socket;
+      /*struct ddsrt_event_file file;  this is for future expansion to be able to register events on files*/
+      /*struct ddsrt_event_ifaddr ifaddr; this is for future expansion to be able to register IP address changes*/
+    } data; /**< Container for the object being watched.*/
+  };
+
+  typedef struct ddsrt_event ddsrt_event_t;
+
+  typedef struct ddsrt_event_queue ddsrt_event_queue_t;
+  
+  /**
+  * @brief Initializes an existing event to indicate a socket.
+  *
+  * @param[out] ev Pointer to the event to initialize.
+  * @param[in] sock Socket to initialize the event with.
+  * @param[in] flags Flags to set for the event.
+  *
+  * @returns DDS_RETCODE_OK if everything went OK.
+  */
+  dds_return_t ddsrt_event_socket_init(ddsrt_event_t* ev, ddsrt_socket_t sock, uint32_t flags);
+
+  /**
+  * @brief Event queue creation function.
+  *
+  * @returns Pointer to the created queue.
+  */
+  ddsrt_event_queue_t* ddsrt_event_queue_create(void);
+
+  /**
+  * @brief Cleans up an event queue.
+  *
+  * Will finish the event queue first, then free the memory of the queue.
+  *
+  * @param[out] queue The queue to clean up.
+  *
+  * @returns DDS_RETCODE_OK if everything went OK.
+  */
+  dds_return_t ddsrt_event_queue_destroy(ddsrt_event_queue_t* queue);
+
+  /**
+  * @brief Getter for the number of stored events.
+  *
+  * @param[in] queue The queue to get the number of events of.
+  *
+  * @returns The number of events stored by the queue.
+  */
+  size_t ddsrt_event_queue_nevents(ddsrt_event_queue_t* queue);
+
+  /**
+  * @brief Triggers a wait for events for the queue.
+  *
+  * @param[out] queue The queue to trigger.
+  * @param[in] reltime The maximum amount of time to wait.
```

Comment:
I added the timeout parameter since both the ddsrt_select and kevent functions allow for a timeout to be set on the function call.
I propose to have the timeout to be truly infinite, not just "very long" (9.2 * 10^9 seconds, which is approx. 300 years) if this parameter is set to be DDS_INFINITY to use the explicit infinite wait time for kevent and select methods.